sdk/core/azure-core/src/main/java/com/azure/core/implementation/RetriableDownloadFlux.java

```diff
@@ -62,7 +62,8 @@ public void subscribe(CoreSubscriber<? super ByteBuffer> actual) {
                 int updatedRetryCount = retryCount + 1;
 
                 if (updatedRetryCount > maxRetries) {
-                    LOGGER.log(LogLevel.ERROR, () -> "Exhausted all retry attempts while downloading.", exception);
+                    LOGGER.log(LogLevel.ERROR, () -> "Exhausted all retry attempts while downloading, "
+                        + updatedRetryCount + " of " + maxRetries + ".", exception);
```

Comment:
If `maxRetries` value is set to 3, this would read "Exhausted all retry attempts while downloading, 4 of 3." which looks odd. We should log it as "3 of 3".